### PR TITLE
Rewrite to use react instead of @yr/component

### PR DESCRIPTION
**Pull request by Saegrov**
* Change to typescript
* Add prettier
* Remove karma-tests

### CODE REVIEW

File: src/lib/connect.tsx [hunk 1]
```diff
@@ -58,37 +54,33 @@ module.exports = function connect(generateProps) {
...
       /**
        * Determine if component should render based on result of 'generateProps'.
        * If not rendering, manually notifies all connected child containers instead of relying on default render flow
-       * @param {Object} nextProps
-       * @param {Object} nextState
-       * @param {Object} nextContext
-       * @returns {Boolean}
        */
-      shouldComponentUpdate(nextProps, nextState, nextContext) {
+      shouldComponentUpdate(_nextProps: any, nextState: any): boolean {
```
Comment (esseb) [inline src/lib/connect.tsx]: Is the underscore for unused parameter a typescript convention?

File: src/lib/select.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
-'use strict';
+import { isEqual } from '@yr/is-equal';
+import { isBrowser } from '@yr/runtime';
 
-const isEqual = require('@yr/is-equal');
-const runtime = require('@yr/runtime');
+export type SomeFn = (selectors: any[], context?: any, props?: any) => any;
```
Comment (esseb) [inline src/lib/select.ts]: I don't understand this name. Can you think of a better name than `SomeFn`?

**Review by esseb — APPROVED** [review]
LGTM

File: src/lib/connect.tsx (hunk 1 above)
Comment (Saegrov) [reply src/lib/connect.tsx]: Yes:
https://github.com/Microsoft/TypeScript/pull/9464

**Pull request merged**